From 69218d5d2854acaa7a11c777244de4a297d2fbb9 Mon Sep 17 00:00:00 2001
From: "Mara Sophie Grosch via groups.io" <littlefox=lf-net.org@groups.io>
Date: Wed, 23 Mar 2022 18:22:34 +0800
Subject: [PATCH 2/2] MdeModulePkg/NvmExpressPei: fix check for NVM command set

Previous commit fixed that check in DXE, this one now for PEI.

Signed-off-by: Mara Sophie Grosch <littlefox@lf-net.org>
Reviewed-by: Hao A Wu <hao.a.wu@intel.com>

Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1007793
Origin: https://github.com/tianocore/edk2/commit/69218d5d2854acaa7a11c777244de4a297d2fbb9
Last-Updated: 2022-03-28

Index: edk2/MdeModulePkg/Bus/Pci/NvmExpressPei/NvmExpressPeiHci.c
===================================================================
--- edk2.orig/MdeModulePkg/Bus/Pci/NvmExpressPei/NvmExpressPeiHci.c
+++ edk2/MdeModulePkg/Bus/Pci/NvmExpressPei/NvmExpressPeiHci.c
@@ -571,7 +571,7 @@ NvmeControllerInit (
   // Read the controller Capabilities register and verify that the NVM command set is supported
   //
   NVME_GET_CAP (Private, &Private->Cap);
-  if (Private->Cap.Css != 0x01) {
+  if ((Private->Cap.Css & BIT0) == 0) {
     DEBUG ((DEBUG_ERROR, "%a: The NVME controller doesn't support NVMe command set.\n", __FUNCTION__));
     return EFI_UNSUPPORTED;
   }
